idutils/utils.py

```diff
@@ -0,0 +1,392 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of IDUtils
+# Copyright (C) 2024 CERN.
+#
+# IDUtils is free software; you can redistribute it and/or modify
+# it under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Utility file containing ID parsers."""
+
+import re
+
+import isbnlib
+
+doi_regexp = re.compile(
```

Comment:
Can we maybe move the relevant regex patterns to a separate file e.g. `regex.py`? We can leave here actual util functions otherwise this will continue becoming a catch-all place....

Reply:
Yes 👍
More options to consider: `patterns.py` or `rules.py`?